lib/core/layout/components/header/header.component.spec.ts

```diff
@@ -0,0 +1,107 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentFixture, TestBed, async } from '@angular/core/testing';
+import { HeaderLayoutComponent } from './header.component';
+import { setupTestBed } from '../../../testing/setupTestBed';
+import { CoreTestingModule } from '../../../testing/core.testing.module';
+import { By } from '@angular/platform-browser';
+import { LayoutModule } from '../..';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+import { ComponentFactoryResolver, Injector, ComponentFactory } from '@angular/core';
+
+describe('HeaderLayoutComponent', () => {
+    let fixture: ComponentFixture<HeaderLayoutComponent>;
+    let component: HeaderLayoutComponent;
+
+    describe('Input parameters', () => {
+        setupTestBed({
+            imports: [CoreTestingModule]
+        });
+
+        beforeEach(() => {
+            fixture = TestBed.createComponent(HeaderLayoutComponent);
+            component = fixture.componentInstance;
+        });
+
+        afterEach(() => {
+            fixture.destroy();
+        });
+
+        it('should create instance of HeaderLayoutComponent', () => {
+            expect(fixture.componentInstance instanceof HeaderLayoutComponent).toBe(true, 'should create HeaderLayoutComponent');
+        });
+
+        it('title element should been displayed', () => {
+            const titleElement = fixture.debugElement.query(By.css('.adf-app-title'));
+            expect(titleElement === null).toBeFalsy();
+        });
+
+        it('should show TEST TITLE', () => {
+            component.title = 'TEST TITLE';
+            fixture.detectChanges();
+
+            const titleElement = fixture.nativeElement.querySelector('.adf-app-title');
+            expect(titleElement.innerText).toEqual('TEST TITLE');
+        });
+
+        it('color attribute should be present on mat-toolbar', () => {
+            component.color = 'primary';
+            fixture.detectChanges();
+
+            const toolbar = fixture.nativeElement.querySelector('mat-toolbar');
+            expect(toolbar.getAttribute('ng-reflect-color') === null).toBeFalsy();
+            expect(toolbar.getAttribute('ng-reflect-color')).toEqual('primary');
+        });
+
+        it('should display the img element with the expected src if a logo path is set', () => {
+            component.logo = 'logo.png';
+            fixture.detectChanges();
+
+            const logo = fixture.nativeElement.querySelector('.adf-app-logo');
+            const src = logo.getAttribute('src');
+            expect(logo === null).toBeFalsy();
+            expect(src).toEqual('logo.png');
+        });
+    });
+});
+
+describe('Template tranclusion', () => {
```

Comment:
I'm not sure ComponentFactoryResolver is needed for test this behaviour. Let's do it together on Monday.